### DISCUSSION TITLE
Problem Create Custom Extension with OASFilter in java

### DISCUSSION THREAD

**Discussion by Zaccher** [smallrye/smallrye-open-api#1380] [Q&A] [answered]
Hi. I need help creating an extension in the .yaml file that is generated with OpenAPI SmallRye.
What I want to achieve is something like this:
```
x-amazon-apigateway-integration:
        uri: "http://${stageVariables.mdwEndpoint}/rest/{version}/factoring/signed-contract"
        responses:
          default:
            statusCode: "200"
            responseTemplates:
              application/json: "{\"json\": \"template\"}"
```

To try and make a trial run, I tried writing this code:

```
Operation apiGatewayIntegrationOperation = OASFactory.createOperation().addExtension("uri", "http://......");
myOperation.addExtension("amazon-apigateway-integration", apiGatewayIntegrationOperation );
```

This code is contained in a class implementing OASFilter, more precisely in a method (@Override) called 
```
filterPathItem(PathItem pathItem){....}
```

However, what I get is the following code:
```
x-amazon-apigateway-integration: null
```

Can someone explain to me how I can build a custom extension that does not just contain a normal String but a complex Object?

Thank you very much for your help.

**Comment by MikeEdgar** [comment, marked as answer]
@Zaccher , try creating your extension as a `Map` instead. Something like this:
```java
Map<String, Object> apiGatewayIntegrationOperation = 
    Map.of("uri", "https://...",
           "responses", Map.of(
               "default", Map.of("stausCode", "200")
           ));
myOperation.addExtension("amazon-apigateway-integration", apiGatewayIntegrationOperation );
```

**Reply by Zaccher** [reply]
Thank you very much. It works!!!! You saved my life!!!!
